subtests/docker_cli/systemd/systemd.py

```diff
@@ -1,86 +1,140 @@
+
+Operational Detail
+----------------------
+
+systemd pull
+~~~~~~~~~~~~~~~~~
+#. Edit unitfile ``docker-pull.service`` & copy it to ``/etc/systemd/system``
+#. Systemd service is started using this file to pull an image.
+#. Systemd service is stoped and image removed
+
+systemd build
+~~~~~~~~~~~~~~~~~
+#. Edit unitfile ``docker-build.service`` & copy it to ``/etc/systemd/system``
+#. Edit Dockerfile
+#. Systemd service is started using this file to build an image.
+#. Systemd service is stoped and image removed
+
+systemd run
+~~~~~~~~~~~~~~~~~
+#. Edit unitfile ``p4321.service`` and copy to ``/etc/systemd/system``
+#. Edit ``Dockerfile`` and copy to test temporary directory
+#. Copy a script ``p4321-server.py`` to test temporary directory
+#. Built an image using the Dockerfile and script
+#. Systemd starts a container using this image
+#. The container writes current time to port ``4321``.
+#. From host, the socket is read and the value is checked
+#. Finally, container is stopped through systemd and system is cleaned up.
+
+Prerequisites
+----------------
+*  Docker daemon is running
+*  systemd unitfiles can be copied to ``/etc/systemd/system``
+*  systemd actions of start/status/stop & daemon-reload can be performed
+*  An image stated in systemd.ini can be pulled & build
 """
 
-import shutil
-import time
-import socket
-from os.path import exists
-from os import unlink
+from os.path import exists, join
+from os import rename, unlink
 from autotest.client import utils
-from autotest.client.shared.utils import is_port_free
-from dockertest.subtest import Subtest
-from dockertest.dockercmd import DockerCmd
+from dockertest.subtest import SubSubtest
+from dockertest.subtest import SubSubtestCaller
 from dockertest.images import DockerImages
-from dockertest.output import mustpass
 from dockertest.xceptions import DockerTestError
 
 
-class systemd(Subtest):
+class systemd(SubSubtestCaller):
+    pass
+
+
+class systemd_base(SubSubtest):
 
     def initialize(self):
-        super(systemd, self).initialize()
-        unit_file_srcpath = '{}{}'.format(self.bindir, '/p4321.service')
-        shutil.copyfile(unit_file_srcpath, self.config['sysd_unitf_dest'])
-        dkrcmd = DockerCmd(self, 'build', ['--force-rm -t',
-                                           self.config['name'], self.bindir])
-        mustpass(dkrcmd.execute())
+        super(systemd_base, self).initialize()
+
+        # Copy unit file into place, with translations.
+        bindir = self.parent_subtest.bindir
+        unitfile = self.config['sysd_unit_file']
+        unitfile_src = join(bindir, unitfile)
+        self.sub_stuff['unitfile_dst'] = join('/etc/systemd/system', unitfile)
+        self.sed_file(unitfile_src, self.sub_stuff['unitfile_dst'])
+
+        # Dockerfile & requirements to tmp workdir (not all tests use this)
+        for f in ['Dockerfile', 'p4321-server.py']:
+            self.sed_file(join(bindir, f), join(self.tmpdir, f))
```

Comment:
Can you explain your concern? I don't understand what you find dangerous. Your comment about "copy to a tmpdir" (which is exactly what this does) makes me wonder if you missed my rewrite of `sed_file`.